MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionPresenter.cpp

```diff
@@ -453,8 +460,188 @@ void EnggDiffractionPresenter::processRebinMultiperiod() {
   startAsyncRebinningPulsesWorker(runNo, nperiods, timeStep, outWSName);
 }
 
+// Fitting Tab Run Number & Bank handling here
+void MantidQt::CustomInterfaces::EnggDiffractionPresenter::
+    fittingRunNoChanged() {
+
+  try {
+    std::string strFocusedFile = m_view->getFittingRunNo();
+    QString focusedFile = QString::fromStdString(strFocusedFile);
+    // file name
+    Poco::Path selectedfPath(strFocusedFile);
+    Poco::Path bankDir;
+
+    // handling of vectors
+    auto runnoDirVector = m_view->getFittingRunNumVec();
+    runnoDirVector.clear();
+
+    std::string strFPath = selectedfPath.toString();
+    // returns empty if no directory is found
+    std::vector<std::string> splitBaseName =
+        m_view->splitFittingDirectory(strFPath);
+    // runNo when single focused file selected
+    std::vector<std::string> runNoVec;
+
+    if (selectedfPath.isFile() && !splitBaseName.empty()) {
+
+#ifdef __unix__
```

Comment:
All the tests pass for me locally. But I have several ENGINX files from testing other PRs. So I'd suspect that it is failing cause it doesn't find an input file. This is the `-V` output that I get:

```
1470: Test command: /usr/bin/cmake "-E" "chdir" "/home/fedemp/test/build-mantid/bin/Testing" "/home/fedemp/test/build-mantid/bin/CustomInterfacesTest" "EnggDiffractionPresenterTest"
1470: Test timeout computed to be: 300
1470: Running 36 testsFrameworkManager-[Notice] Welcome to Mantid 3.6.20160503.1353
1470: FrameworkManager-[Notice] Please cite: http://dx.doi.org/10.1016/j.nima.2014.07.029 and this release: http://dx.doi.org/10.5286/Software/Mantid
1470: .....EngineeringDiffractionGUI-[Notice] EnggDiffraction GUI: starting new calibration. This may take a few seconds... 
1470: EngineeringDiffractionGUI-[Notice] Generating new calibration file: ENGINX_8899999988_9999999999_both_banks.prm
1470: EngineeringDiffractionGUI-[Notice] Calculating Vanadium corrections. This may take a few seconds...
1470: EngineeringDiffractionGUI-[Error] Failed to calculate Vanadium corrections. There was an error in the execution of the algorithms required to calculate Vanadium corrections. Some properties passed to the algorithms were invalid. This is possibly because some of the settings are not consistent. Please check the log messages for details. Details: Invalid value for property Filename (list of str lists) "8899999988": Unable to find file matching the string "IMAT00000", even after appending suggested file extensions.
1470: EngineeringDiffractionGUI-[Error] The calibration calculations failed. Some input properties were not valid. See log messages for details. 
1470: EngineeringDiffractionGUI-[Warning] The cablibration did not finish correctly.
1470: ....EngineeringDiffractionGUI-[Notice] EnggDiffraction GUI: starting cropped calibration. This may take a few seconds... 
1470: EngineeringDiffractionGUI-[Notice] Generating new calibration file: UNKNOWNINST_8899999988_9999999999_calibration.prm
1470: EngineeringDiffractionGUI-[Notice] Calculating Vanadium corrections. This may take a few seconds...
1470: EngineeringDiffractionGUI-[Error] Failed to calculate Vanadium corrections. There was an error in the execution of the algorithms required to calculate Vanadium corrections. Some properties passed to the algorithms were invalid. This is possibly because some of the settings are not consistent. Please check the log messages for details. Details: Invalid value for property Filename (list of str lists) "8899999988": Unable to find file matching the string "IMAT00000", even after appending suggested file extensions.
1470: EngineeringDiffractionGUI-[Error] The calibration calculations failed. Some input properties were not valid. See log messages for details. 
1470: EngineeringDiffractionGUI-[Warning] The cablibration did not finish correctly.
1470: .EngineeringDiffractionGUI-[Notice] EnggDiffraction GUI: starting cropped calibration. This may take a few seconds... 
1470: EngineeringDiffractionGUI-[Notice] Generating new calibration file: UNKNOWNINST_8899999988_9999999999_calibration.prm
1470: EngineeringDiffractionGUI-[Notice] Calculating Vanadium corrections. This may take a few seconds...
1470: EngineeringDiffractionGUI-[Error] Failed to calculate Vanadium corrections. There was an error in the execution of the algorithms required to calculate Vanadium corrections. Some properties passed to the algorithms were invalid. This is possibly because some of the settings are not consistent. Please check the log messages for details. Details: Invalid value for property Filename (list of str lists) "8899999988": Unable to find file matching the string "IMAT00000", even after appending suggested file extensions.
1470: EngineeringDiffractionGUI-[Error] The calibration calculations failed. Some input properties were not valid. See log messages for details. 
1470: EngineeringDiffractionGUI-[Warning] The cablibration did not finish correctly.
1470: .....EngineeringDiffractionGUI-[Error] Error in specification of banks found when starting the focusing process. Cannot continue.
1470: .........EngineeringDiffractionGUI-[Notice] EnggDiffraction GUI: starting new pre-processing (re-binning) with a TOF bin into workspace 'engggui_preproc_time_ws'. This may take some seconds... 
1470: Load-[Notice] Load started
1470: Load-[Notice] Load successful, Duration 0.31 seconds
1470: Rebin-[Notice] Rebin started
1470: Rebin-[Notice] Rebin successful, Duration 3.94 seconds
1470: EngineeringDiffractionGUI-[Notice] Pre-processing (re-binning) finished - the output workspace is ready.
1470: ..........EngineeringDiffractionGUI-[Error] Changing instrument is not supported!
1470: ..OK!
1/1 Test #1470: CustomInterfacesTest_EnggDiffractionPresenterTest ...   Passed    6.89 sec
```